openedx/openedx-platform
# feat: [FC-0044] XBlock's children API as DRF

**Settings**

```yaml
TUTOR_GROVE_WAFFLE_FLAGS:
  - name: contentstore.new_studio_mfe.use_new_unit_page
    everyone: true

TUTOR_GROVE_MFE_LMS_COMMON_SETTINGS:
  MFE_CONFIG:
    ENABLE_UNIT_PAGE: true
```

## Description
This PR addresses a need in the development of the Course Unit Page feature within the **frontend-app-course-authoring** repository. The core objective is the implement of a DRF endpoint designed to render and deliver children data for container xblock.

This PR also includes:
- adding unit type to `create_xblock_info` handler. It needs to display type of unit icons for the unit course page. Each unit must have one of the following types: `video`, `vertical`, `problem`, `lock`.
- resolving problem with 500 error if user adds a Content Experiment (also needs to discuss)

## Testing instructions
1. Run master devstack.
2. Start platform `make dev.up.lms+cms` and make checkout on this branch.
3. Go to `http://localhost:18010/api-docs`.
4. Find the required API endpoint `/api/contentstore/v1/container/vertical/{usage_key_string}/children`.
5. Open needed unit page in cms `http://localhost:18010/container/{usage_key_string}` and make sure nothing has changed.
6. Copy `usage_key_string` from previous step and paste it to required API endpoint to get data for unit.

## How to reproduce an error 
The problem arose at the stage of defining icons for children xblock's of the unit.
Just when we tried to get `get_icon()` for xblock which have `split_test` children.

1. Go to Advanced settings page and add "split_test" to Advanced module list
2. Go to Group configurations page and create group in Experiment group configurations section
3. Go to Course Outline, create new unit, go to unit page
4. In the Advanced select "Content Experiment"

```
File "/edx/app/edxapp/edx-platform/xmodule/vertical_block.py", line 254, in get_icon_class
    child_classes = {child.get_icon_class() for child in self.get_children()}
  File "/edx/app/edxapp/edx-platform/xmodule/vertical_block.py", line 254, in <setcomp>
    child_classes = {child.get_icon_class() for child in self.get_children()}
  File "/edx/app/edxapp/edx-platform/xmodule/split_test_block.py", line 398, in get_icon_class
    return self.child.get_icon_class() if self.child else 'other'
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/edx/app/edxapp/edx-platform/xmodule/split_test_block.py", line 176, in child
    if self.child_block is not None:
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/edx/app/edxapp/edx-platform/xmodule/split_test_block.py", line 166, in child_block
    child_blocks = self.get_child_blocks()
  File "/edx/app/edxapp/edx-platform/xmodule/split_test_block.py", line 216, in get_child_blocks
    group_id = self.get_group_id()
  File "/edx/app/edxapp/edx-platform/xmodule/split_test_block.py", line 243, in get_group_id
    user_service = self.runtime.service(self, 'user')
  File "/edx/app/edxapp/edx-platform/xmodule/modulestore/split_mongo/caching_descriptor_system.py", line 245, in service
    return super().service(block, service_name)
  File "/edx/app/edxapp/edx-platform/xmodule/x_module.py", line 1502, in service
    service = super().service(block=block, service_name=service_name)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/xblock/runtime.py", line 1125, in service
    raise NoSuchServiceError(f"Service {service_name!r} is not available.")
xblock.exceptions.NoSuchServiceError: Service 'user' is not available.
```
